From e68162e983d278affb3641c897a6497385491921 Mon Sep 17 00:00:00 2001
From: "hayim@marvell.com" <hayim@marvell.com>
Date: Thu, 28 Jan 2016 13:54:54 +0200
Subject: [PATCH 0856/1240] dts: add debug option for printing dtb file (after
 modify) during Linux boot

- Before Linux boot, u-boot updates the dtb file.
- There is not option to check it before or during boot time.
- This patch add the option (by using compilation flag) to
  print it after it's loaded and updated.

Change-Id: I1dbdf43edb3db290b3d5f133593a467562135617
Signed-off-by: hayim@marvell.com <hayim@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27061
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 common/cmd_fdt.c      | 3 +--
 common/image-fdt.c    | 5 +++++
 include/fdt_support.h | 2 ++
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/common/cmd_fdt.c b/common/cmd_fdt.c
index dc59fab..fb34ad4 100644
--- a/common/cmd_fdt.c
+++ b/common/cmd_fdt.c
@@ -30,7 +30,6 @@ DECLARE_GLOBAL_DATA_PTR;
 
 static int fdt_valid(struct fdt_header **blobp);
 static int fdt_parse_prop(char *const*newval, int count, char *data, int *len);
-static int fdt_print(const char *pathp, char *prop, int depth);
 static int is_printable_string(const void *data, int len);
 
 /*
@@ -903,7 +902,7 @@ static void print_data(const void *data, int len)
  * Recursively print (a portion of) the working_fdt.  The depth parameter
  * determines how deeply nested the fdt is printed.
  */
-static int fdt_print(const char *pathp, char *prop, int depth)
+int fdt_print(const char *pathp, char *prop, int depth)
 {
 	static char tabs[MAX_LEVEL+1] =
 		"\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t"
diff --git a/common/image-fdt.c b/common/image-fdt.c
index e3f06cd..f9de8f2 100644
--- a/common/image-fdt.c
+++ b/common/image-fdt.c
@@ -520,6 +520,11 @@ int image_setup_libfdt(bootm_headers_t *images, void *blob,
 		ft_board_setup_ex(blob, gd->bd);
 #endif
 
+#if defined(CONFIG_FDT_DEBUG_PRINT)
+	set_working_fdt_addr(blob);
+	fdt_print("/", NULL, 32 /* Max depth */);
+#endif
+
 	return 0;
 err:
 	printf(" - must RESET the board to recover.\n\n");
diff --git a/include/fdt_support.h b/include/fdt_support.h
index 1f19fe4..56e60d5 100644
--- a/include/fdt_support.h
+++ b/include/fdt_support.h
@@ -175,6 +175,8 @@ int arch_fixup_memory_node(void *blob);
 int fdt_setup_simplefb_node(void *fdt, int node, u64 base_address, u32 width,
 			    u32 height, u32 stride, const char *format);
 
+int fdt_print(const char *pathp, char *prop, int depth);
+
 #endif /* ifdef CONFIG_OF_LIBFDT */
 
 #ifdef USE_HOSTCC
-- 
1.9.1

